Marlin/src/feature/TMC2660.cpp

```diff
@@ -0,0 +1,1030 @@
+/*
+ TMC26XStepper.cpp - - TMC26X Stepper library for Wiring/Arduino
+ 
+ based on the stepper library by Tom Igoe, et. al.
+ 
+ Copyright (c) 2011, Interactive Matter, Marcus Nowotny
+ 
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+ 
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+ 
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ 
+ */
+
+//#include "Arduino.h"
+
+#include <SPI.h>
+#include "TMC2660.h"
+
+#include "../HAL/HAL_STM32F7/HAL_STM32F7.h"
```

Comment:
You can't access HAL headers directly like this from Marlin base code, would need routed through the HAL layer, but if this is needed here then this lib needs moved to your HAL folder as it is not generic.

Reply:
Will check if i really need it, right now i cant test compile because of some change that happen'd or did not happen when multi-host was merged..

But had no problems compiling it before like this, no warnings..

Couldn't use the normal TCM26x lib since i needed to modify it slightly so to not break anything that uses the normal TCM26x i did it this way..

Waiting for @teemuatlut and his new TCM2660 library and will use that when he has time to finnish it :)

The STM32Generic core doesn't like the old style spi setup, i need to use:
STEPPER_SPI.beginTransaction(SPISettings(4000000, MSBFIRST, SPI_MODE3)); 

That's one of the changes in the TCM26x files..

Reply:
You can already use TMCStepper library for setting up the registers. It's mostly the helper methods that are missing.
There will also come a point when both these files become unnecessary and "all" the TMC stepper support would be handled by the universal library.

Reply:
This is why it's not passing Travis. As @p3p pointed out, you shouldn't be accessing your *HAL.h* directly from within a .cpp file. Because this file isn't wrapped with `#ifdef STM32F7`, it will try to be compiled in every configuration, which means that every configuration will include your *HAL_STM32F7.h* file.

For now, the easiest thing to do would be to go ahead and wrap this file in the `#ifdef STM32F7` guard.

Reply:
yup, moved the TCM2660.x files to the HAL dir and gave it a #ifdef STM32F7 guard, passes Travis nicely now :)

Github seems to have some problems with messaging today, i get a email notification almost an hour after someone writes here...

Reply:
@teemuatlut "universal library" is going to be so nice, thanks for your work :)